active_plugins/runilastik.py

```diff
@@ -0,0 +1,247 @@
+#################################
+#
+# Imports from useful Python libraries
+#
+#################################
+
+import os
+import subprocess
+import shutil
+import uuid
+import logging
+import sys
+import h5py 
+import tempfile
+
+
+#################################
+#
+# Imports from CellProfiler
+#
+##################################
+
+from cellprofiler_core.image import Image
+from cellprofiler_core.module import ImageProcessing
+from cellprofiler_core.setting.choice import Choice
+from cellprofiler_core.preferences import get_default_output_directory
+from cellprofiler_core.setting.text import (
+    Directory,
+    Filename,
+    Pathname,
+)
+
+ilastik_link = "https://doi.org/10.1038/s41592-019-0582-9"
+LOGGER = logging.getLogger(__name__)
+
+
+__doc__ = """\
+Runilastik
+=======
+
+Use an ilastik pixel classifier to generate a probability image. Each
+channel represents the probability of the pixels in the image belong to
+a particular class. Use **ColorToGray** to separate channels for further
+processing. For example, use **IdentifyPrimaryObjects** on a
+(single-channel) probability map to generate a segmentation. The order
+of the channels in **ColorToGray** is the same as the order of the
+labels within the ilastik project.
+
+Additionally, please ensure CellProfiler is configured to load images in
+the same format as ilastik. For example, if your ilastik classifier is
+trained on RGB images, use **NamesAndTypes** to load images as RGB by
+selecting "*Color image*" from the *Select the image type* dropdown. If
+your classifier expects grayscale images, use **NamesAndTypes** to load
+images as "*Grayscale image*".
+
+Add more documentation.
+"""
+
+ILASTIK_DOCKER = "biocontainers/ilastik:1.4.0_cv2"
+
+class Runilastik(ImageProcessing):
+    module_name = "Runilastik"
+
+    variable_revision_number = 1  
+
+    doi = {
+        "Please cite the following when using Runilastik:": "https://doi.org/10.1038/s41592-019-0582-9",
+    }
+
+    def create_settings(self):
+        super(Runilastik, self).create_settings()
+
+        self.docker_or_local = Choice(
+            text="Run ilastik in docker or local environment",
+            choices=["Docker", "Local"],
+            value="Docker",
+            doc="""\
+If Docker is selected, ensure that Docker Desktop is open and running on your
```

Comment:
@sugan89 we should add a warning here about slowness; we should also add warnings here and in the intro re: analysis mode not working on Windows. We may even want to add a setting validator (aka make a warning icon pop up in the pipeline panel) that warns the user if they give a .exe as the executable that they can't use the GUI-analysis mode.

Reply:
We also want to add the link to the version of the Docker we're using!